src/auto-evo/AutoEvoRun.cs

```diff
@@ -174,6 +178,44 @@ public void Start()
         started = true;
     }
 
+    public void OneStep()
+    {
+        if (Running)
+            return;
+
+        started = true;
+
+        var timer = new Stopwatch();
+        timer.Start();
+
+        Running = true;
+
+        try
+        {
+            if (Step())
+                Finished = true;
+        }
+        catch (Exception e)
+        {
+            Aborted = true;
+            GD.PrintErr("Auto-evo failed with an exception: ", e);
+        }
+
+        Running = false;
+
+        RunDuration += timer.Elapsed;
+    }
+
+    public void Continue()
+    {
+        if (Running)
```

Comment:
Isn't there a possible race condition here? As in Running won't be set to true until another thread does that, so if this method is called twice very quickly in a row it can result in multiple auto-evo starts.
Maybe it's enough to set `Running = true` on line 213?